src/pubsub.c

```diff
@@ -235,6 +245,15 @@ dict* getClientPubSubShardChannels(client *c) {
     return c->pubsubshard_channels;
 }
 
+dict **getServerPubSubChannels(unsigned int slot) {
+    UNUSED(slot);
+    return &server.pubsub_channels;
+}
+
+dict **getServerPubSubShardChannels(unsigned int slot) {
+    return &server.pubsubshard_channels[server.cluster_enabled ? slot : 0];
```

Comment:
Can this be called with a non-zero slot if cluster is disabled? If not, we can do an assert instead:

```suggestion
    serverAssert(server.cluster_enabled || slot == 0);
    return &server.pubsubshard_channels[slot];
```

Reply:
No, if cluster is disabled, there will be only one slot, the same as dicts in DBs.